src/Adapter/MSTest.CoreAdapter/Execution/TestMethodRunner.cs

```diff
@@ -345,35 +357,81 @@ internal UnitTestResult[] RunTestMethod()
                 this.testMethodInfo.TestMethodName);
             }
 
-            if (results != null && results.Count > 0)
-            {
-                // aggregate for data driven tests
-                UTF.UnitTestOutcome aggregateOutcome = UTF.UnitTestOutcome.Passed;
+            parentResultWatch.Stop();
+            parentResult.Duration = parentResultWatch.Elapsed;
 
-                foreach (var result in results)
-                {
-                    if (result.Outcome != UTF.UnitTestOutcome.Passed)
-                    {
-                        if (aggregateOutcome != UTF.UnitTestOutcome.Failed)
-                        {
-                            if (result.Outcome == UTF.UnitTestOutcome.Failed
-                                || aggregateOutcome != UTF.UnitTestOutcome.Timeout)
-                            {
-                                aggregateOutcome = result.Outcome;
-                            }
-                        }
-                    }
-                }
+            // Get aggregate outcome.
+            var aggregateOutcome = this.GetAggregateOutcome(results);
+            this.testContext.SetOutcome(aggregateOutcome);
 
-                this.testContext.SetOutcome(aggregateOutcome);
+            // Set a result in case no result is present.
+            if (!results.Any())
+            {
+                results.Add(new UTF.TestResult() { Outcome = aggregateOutcome, TestFailureException = new TestFailedException(UnitTestOutcome.Error, Resource.UTA_NoTestResult) });
             }
-            else
+
+            // In case of data driven, set parent info in results.
+            if (isDataDriven)
             {
-                this.testContext.SetOutcome(UTF.UnitTestOutcome.Unknown);
-                results.Add(new UTF.TestResult() { Outcome = UTF.UnitTestOutcome.Unknown, TestFailureException = new TestFailedException(UnitTestOutcome.Error, Resource.UTA_NoTestResult) });
+                parentResult.Outcome = aggregateOutcome;
+                results = this.UpdateResultsWithParentInfo(results, parentResult);
             }
 
             return results.ToArray().ToUnitTestResults();
         }
+
+        /// <summary>
+        /// Gets aggregate outcome.
+        /// </summary>
+        /// <param name="results">Results.</param>
+        /// <returns>Aggregate outcome.</returns>
+        private UTF.UnitTestOutcome GetAggregateOutcome(List<UTF.TestResult> results)
+        {
+            // In case results are not present, set outcome as unknown.
+            if (!results.Any())
```

Comment:
In case of list, using Count or Any doesn't matter. But in case of Enumerable any is preferred over count as count iterates through entire list. Thus as a general practice, for both list and enumerable I prefer any.